spec/ruby/core/method/shared/to_s.rb

```diff
@@ -31,4 +31,22 @@
   it "returns a String containing the Module the method is referenced from" do
     @string.should =~ /MethodSpecs::MySub/
   end
+
+  ruby_version_is '2.8' do
+    it "returns a String containing the Module containing the method if object has a singleton class but method is not defined in the singleton class" do
+      obj = MethodSpecs::MySub.new
+      obj.singleton_class
+      @m = obj.method(:bar)
+      @string = @m.send(@method).sub(/0x\w+/, '0xXXXXXX')
+      @string.should =~ /MethodSpecs::MySub\(MethodSpecs::MyMod\)/
```

Comment:
Could we be more precise and just check the exact result? (same below)

Reply:
Sure, I'll make that change.  It will make the new specs inconsistent with the existing specs, though.

Reply:
I tried that, but the path to the file and line number is included in the spec output, which would have to be masked out similar to the object id masking.  I really don't think it is a good idea to do that.  I will add a commit that uses a more complete regexp, though.

Reply:
Thanks, it looks good.
I'll try to make a pass over the specs in this file to improve them.